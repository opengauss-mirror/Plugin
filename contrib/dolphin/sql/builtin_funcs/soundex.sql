drop database if exists db_soundex;
create database db_soundex dbcompatibility 'B';
\c db_soundex

select soundex('abc');
select soundex(10);
select soundex('afdsbfdlsafs');
select soundex('dslfaalskf222122');
select soundex('dadfa353dadf333ff');
select soundex('hell');

select 'ibc' sounds like 'abc';
select 'aws' sounds like '10';
select 'aftcbtlxpz' sounds like 'afdsbfdlsafs';
select 'zie' sounds like 'dslfaalskf222122';
select 'adfidf' sounds like 'dadfa353dadf333ff';
select 'hill' sounds like 'hell';
select 'ä¸­Adfab' sounds like '@adp';
select 1 sounds like 2;
select true sounds like 3;
select 2.2 sounds like false;

\c postgres
drop database if exists db_soundex;
