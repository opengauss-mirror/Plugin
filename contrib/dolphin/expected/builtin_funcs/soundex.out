drop database if exists db_soundex;
NOTICE:  database "db_soundex" does not exist, skipping
create database db_soundex dbcompatibility 'B';
\c db_soundex
select soundex('abc');
 soundex 
---------
 A120
(1 row)

select soundex(10);
 soundex 
---------
 
(1 row)

select soundex('afdsbfdlsafs');
  soundex   
------------
 A132134212
(1 row)

select soundex('dslfaalskf222122');
 soundex 
---------
 D241421
(1 row)

select soundex('dadfa353dadf333ff');
 soundex 
---------
 D131
(1 row)

select soundex('hell');
 soundex 
---------
 H400
(1 row)

select soundex('测试');
 soundex 
---------
 测000
(1 row)

select soundex('dbakbdasjcbsdk,kdsjbcksd,cnsk');
      soundex       
--------------------
 D12132123232123252
(1 row)

select soundex('!!!jhdgj');
 soundex 
---------
 J320
(1 row)

select soundex('!!!!!！！jhdgj');
 soundex 
---------
 ！232
(1 row)

select soundex(null);
 soundex 
---------
 
(1 row)

select soundex('12,xa');
 soundex 
---------
 X000
(1 row)

select soundex('dkh测试fkdsjs12');
 soundex 
---------
 D21232
(1 row)

select soundex(false),soundex(true);
 soundex | soundex 
---------+---------
         | 
(1 row)

select soundex('颜'),soundex('谚');
 soundex | soundex 
---------+---------
 颜000   | 谚000
(1 row)

select soundex('颜 '),soundex('谚0');
 soundex | soundex 
---------+---------
 颜000   | 谚000
(1 row)

select soundex('遥'),soundex('摇'),soundex('瑶');
 soundex | soundex | soundex 
---------+---------+---------
 遥000   | 摇000   | 瑶000
(1 row)

select soundex('遥!@#@$#'),soundex('摇0.00001'),soundex('瑶nhgchgh');
 soundex | soundex | soundex 
---------+---------+---------
 遥000   | 摇000   | 瑶520
(1 row)

select soundex('人'),soundex('入');
 soundex | soundex 
---------+---------
 人000   | 入000
(1 row)

select soundex('00人'),soundex('a00入'),soundex('baa入'),soundex('0b0入');
 soundex | soundex | soundex | soundex 
---------+---------+---------+---------
 人000   | A000    | B000    | B000
(1 row)

select soundex('0000食不果腹'),soundex('食不裹腹');
 soundex | soundex 
---------+---------
 食000   | 食000
(1 row)

select soundex('气概'),soundex('气慨');
 soundex | soundex 
---------+---------
 气000   | 气000
(1 row)

select soundex('辐射'),soundex('幅射');
 soundex | soundex 
---------+---------
 辐000   | 幅000
(1 row)

select soundex(b'111');
 soundex 
---------
 
(1 row)

select 'ibc' sounds like 'abc';
 soundex_difference 
--------------------
                  0
(1 row)

select 'aws' sounds like '10';
 soundex_difference 
--------------------
                  0
(1 row)

select 'aftcbtlxpz' sounds like 'afdsbfdlsafs';
 soundex_difference 
--------------------
                  1
(1 row)

select 'zie' sounds like 'dslfaalskf222122';
 soundex_difference 
--------------------
                  0
(1 row)

select 'adfidf' sounds like 'dadfa353dadf333ff';
 soundex_difference 
--------------------
                  0
(1 row)

select 'hill' sounds like 'hell';
 soundex_difference 
--------------------
                  1
(1 row)

select '中Adfab' sounds like '@adp';
 soundex_difference 
--------------------
                  0
(1 row)

select 1 sounds like 2;
 soundex_difference 
--------------------
                  1
(1 row)

select true sounds like 3;
ERROR:  function pg_catalog.soundex_difference(boolean, integer) does not exist
LINE 1: select true sounds like 3;
                    ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
CONTEXT:  referenced column: soundex_difference
select 2.2 sounds like false;
ERROR:  function pg_catalog.soundex_difference(numeric, boolean) does not exist
LINE 1: select 2.2 sounds like false;
                   ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
CONTEXT:  referenced column: soundex_difference
select soundex('и мърморене');
 soundex 
---------
 и000
(1 row)

select soundex('ɔnɛkona mws');
 soundex 
---------
 ɔ5252
(1 row)

select soundex('ш');
 soundex 
---------
 ш000
(1 row)

select soundex('я такая шчасліваяwjdkadskdjk');
 soundex  
----------
 я2323232
(1 row)

select soundex('测T测h测试o测masёння я такая шчасліваяhello');
 soundex 
---------
 测3524
(1 row)

\c postgres
drop database if exists db_soundex;
